feat(apple): MetricKit GA (#8186)
getsentry/sentry-docs: src/platforms/apple/common/configuration/metric-kit.mdx

---
title: MetricKit
sidebar_order: 20
description: "This feature, once enabled, subscribes to MXDiagnosticPayload data of MetricKit and it to Sentry."
---

The [MetricKit](https://developer.apple.com/documentation/metrickit) integration subscribes to [MXHangDiagnostic](https://developer.apple.com/documentation/metrickit/mxhangdiagnostic), [MXDiskWriteExceptionDiagnostic](https://developer.apple.com/documentation/metrickit/mxdiskwriteexceptiondiagnostic)
and [MXCPUExceptionDiagnostic](https://developer.apple.com/documentation/metrickit/mxcpuexceptiondiagnostic), which it converts to Sentry events.

This feature is available on Cocoa 8.14.0 and up. The SDK supports this feature from iOS 15 and up and macOS 12 and up because in these versions, MetricKit delivers diagnostic reports immediately, which allows the Sentry SDK to apply current data from the scope.

> Note that the Cocoa SDK has no control over the stacktraces provided by MetricKit. Some only offer a couple of frames. If these aren't useful to you, you can drop them in <PlatformLink to="/configuration/filtering/#filtering-error-events/">`beforeSend`</PlatformLink>.

Reviewer: ```suggestion
Note that the Cocoa SDK has no control over the stack traces provided by MetricKit. Some only offer a couple of frames. If these aren't useful to you, you can drop them in <PlatformLink to="/configuration/filtering/#filtering-error-events/">`beforeSend`</PlatformLink>.
```

description: "This feature, once enabled, subscribes to MXDiagnosticPayload data of MetricKit and it to Sentry."
---

> Once enabled, this feature subscribes to [MetricKit's](https://developer.apple.com/documentation/metrickit) [MXDiagnosticPayload](https://developer.apple.com/documentation/metrickit/mxdiagnosticpayload) data, converts it to events, and sends it to Sentry. This feature is available on Cocoa 8.14.0 and up.

Reviewer 2: ```suggestion
The MetricKit integration subscribes to [MetricKit's](https://developer.apple.com/documentation/metrickit) [MXDiagnosticPayload](https://developer.apple.com/documentation/metrickit/mxdiagnosticpayload) data, converts it to events, and sends it to Sentry. This feature is available on Cocoa 8.14.0 and up.
```

Reviewer 2: We talk about MetricKit subscribing to MXDiagnosticPayload data in the intro paragraph and then have the list of what the integration subscribes to (MXHangDiagnostic etc.). I'm wondering if it's possible to combine these 2 somehow? It reads odd to have them separated.

- [MXCPUExceptionDiagnostic](https://developer.apple.com/documentation/metrickit/mxcpuexceptiondiagnostic)

The SDK supports this feature from iOS 15 and up and macOS 12 and up because in these versions, MetricKit delivers diagnostic reports immediately, which allows the Sentry SDK to apply current data from the scope.

> It is worth noting that the Cocoa SDK has no control over the stacktraces MetricKit provides. Some stacktraces from MetricKit only offer a couple of frames. You can drop such events in <PlatformLink to="/configuration/filtering/#filtering-error-events/">`beforeSend`</PlatformLink> and identify MetricKit events by looking at the `mechanism.type` of the `exception`, which are `mx_hang_diagnostic`, `mx_cpu_exception`, and `mx_disk_write_exception`.

Reviewer 3: ```suggestion
Note that the Cocoa SDK has no control over the stacktraces provided by MetricKit. Some stacktraces from MetricKit only offer a couple of frames. If these are not useful to you, you can drop such events in <PlatformLink to="/configuration/filtering/#filtering-error-events/">`beforeSend`</PlatformLink>. You can identify MetricKit events by looking at the `mechanism.type` of the `exception`, which are `mx_hang_diagnostic`, `mx_cpu_exception`, and `mx_disk_write_exception`.
```

> The SDK supports this feature from iOS 15 and up and macOS 12 and up because in these versions, MetricKit delivers diagnostic reports immediately, which allows the Sentry SDK to apply current data from the scope.
> Note that the Cocoa SDK has no control over the stacktraces provided by MetricKit. Some stacktraces from MetricKit only offer a couple of frames. If these are not useful to you, you can drop such events in <PlatformLink to="/configuration/filtering/#filtering-error-events/">`beforeSend`</PlatformLink>. You can identify MetricKit events by looking at the `mechanism.type` of the `exception`, which are `mx_hang_diagnostic`, `mx_cpu_exception`, and `mx_disk_write_exception`.

Reviewer 2: ```suggestion
The SDK supports this feature from iOS 15 and up and macOS 12 and up because in these versions, MetricKit delivers diagnostic reports immediately, which allows the Sentry SDK to apply current data from the scope.

Note that the Cocoa SDK has no control over the stacktraces provided by MetricKit. Some only offer a couple of frames. If these aren't useful to you, you can drop them in <PlatformLink to="/configuration/filtering/#filtering-error-events/">`beforeSend`</PlatformLink>. 

You can identify MetricKit events by looking at the `mx_hang_diagnostic`, `mx_cpu_exception`, and `mx_disk_write_exception` `mechanism.type`.
```

> This feature requires opt-in, and can be enabled by setting the `enableMetricKit` option to `true`:

Reviewer 3: ```suggestion
To use this feature, enable it by setting the `enableMetricKit` option to `true`:
```

> To use this feature, enable it by setting the `enableMetricKit` option to `true`:

Reviewer 2: ```suggestion
Enable this feature by setting the `enableMetricKit` option to `true`:
```
